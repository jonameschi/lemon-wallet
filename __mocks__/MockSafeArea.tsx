import React, { FC, PropsWithChildren } from 'react';

import { Metrics, SafeAreaProvider } from 'react-native-safe-area-context';

const MOCK_INITIAL_METRICS: Metrics = {
  frame: {
    width: 320,
    height: 640,
    x: 0,
    y: 0,
  },
  insets: {
    left: 0,
    right: 0,
    bottom: 0,
    top: 0,
  },
};

type MockSafeAreaContextProps = {
  initialWindowMetrics?: Metrics | null;
};

const MockSafeAreaContext: FC<PropsWithChildren<MockSafeAreaContextProps>> = ({
  children,
  initialWindowMetrics,
}) => (
  <SafeAreaProvider
    initialMetrics={initialWindowMetrics || MOCK_INITIAL_METRICS}>
    {children}
  </SafeAreaProvider>
);

export { MockSafeAreaContext };
export type { MockSafeAreaContextProps };
