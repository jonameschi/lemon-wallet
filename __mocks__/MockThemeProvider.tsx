import { FC, PropsWithChildren } from 'react';
import { ThemeProvider } from 'styled-components/native';
import { theme } from 'theme/theme';

export const MockThemeProvider: FC<PropsWithChildren> = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);
