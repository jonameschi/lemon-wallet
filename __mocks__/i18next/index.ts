const useMock = (key: string): string => key;

export default {
  t: useMock,
};
