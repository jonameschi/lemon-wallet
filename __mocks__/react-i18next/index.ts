/* eslint-disable import/no-import-module-exports */
import { I18nextProvider, initReactI18next } from 'react-i18next';

const useMock = {
  t: (key: string, options: object | undefined): string =>
    options ? `${key} - ${JSON.stringify(options)}` : key,
};

module.exports = {
  useTranslation: () => useMock,
  I18nextProvider,
  initReactI18next,
};
