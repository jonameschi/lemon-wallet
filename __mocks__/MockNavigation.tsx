import React, { FunctionComponent, PropsWithChildren } from 'react';

import { NavigationContext } from '@react-navigation/native';
import { MockSafeAreaContext, MockSafeAreaContextProps } from './MockSafeArea';
import { MockThemeProvider } from './MockThemeProvider';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const mockNavigationProps: any = {
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
  isFocused: () => true,
  addListener: jest.fn(() => () => ({ remove: jest.fn() })),
};

type MockNavigationProps = PropsWithChildren<MockSafeAreaContextProps>;

const MockNavigation: FunctionComponent<MockNavigationProps> = ({
  children,
  initialWindowMetrics,
}) => (
  <MockSafeAreaContext initialWindowMetrics={initialWindowMetrics}>
    <NavigationContext.Provider value={mockNavigationProps}>
      <MockThemeProvider>{children}</MockThemeProvider>
    </NavigationContext.Provider>
  </MockSafeAreaContext>
);

export { MockNavigation, mockNavigationProps };
