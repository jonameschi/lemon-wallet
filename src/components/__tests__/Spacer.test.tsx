import { ComponentProps } from 'react';
import { Spacer } from '@components';
import { render } from '@testing-library/react-native';
import { MockThemeProvider } from '../../../__mocks__/MockThemeProvider';

const renderTree = (props: ComponentProps<typeof Spacer>) =>
  render(
    <MockThemeProvider>
      <Spacer {...props} />
    </MockThemeProvider>,
  );

describe('Spacer component', () => {
  it('should render a snapshot - with horizontal orientation and size 0', () => {
    const { toJSON } = renderTree({
      orientation: 'horizontal',
      size: 0,
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render a snapshot - without vertical orientation and size 12', () => {
    const { toJSON } = renderTree({ orientation: 'vertical', size: 12 });
    expect(toJSON()).toMatchSnapshot();
  });
});
