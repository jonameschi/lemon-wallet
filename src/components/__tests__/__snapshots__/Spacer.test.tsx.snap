// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer component should render a snapshot - with horizontal orientation and size 0 1`] = `
<View
  orientation="horizontal"
  size={0}
  style={
    {
      "width": 0,
    }
  }
/>
`;

exports[`Spacer component should render a snapshot - without vertical orientation and size 12 1`] = `
<View
  orientation="vertical"
  size={12}
  style={
    {
      "height": 12,
    }
  }
/>
`;
